java/java-analysis-impl/src/com/intellij/codeInspection/dataFlow/CustomMethodHandlers.java

```diff
@@ -193,7 +193,14 @@ private static CustomMethodHandler toValue(DfTypeCustomMethodHandler handler) {
                            "newCopyOnWriteArraySet", "newConcurrentHashSet", "newTreeSet").parameterCount(0),
                 staticCall("com.google.common.collect.Maps", "newHashMap", "newLinkedHashMap", "newIdentityHashMap",
                            "newConcurrentHashMap", "newTreeMap").parameterCount(0)),
-              toValue((arguments, state, factory, method) -> COLLECTION_SIZE.asDfType(intValue(0)).meet(LOCAL_OBJECT)));
+              toValue((arguments, state, factory, method) -> COLLECTION_SIZE.asDfType(intValue(0)).meet(LOCAL_OBJECT)))
+    //for propagation CONSUMED_STREAM
+    .register(ConsumedStreamUtils.getSkipStreamMatchers(), (args, memState, factory, method) -> args.myQualifier)
```

Comment:
Probably providing a `-> this` contract would produce the same result? Though it's not exactly specified that it will be so...

Reply:
Moved to HardcodedContract. Honestly, I don't want to work with External annotation, because as you wrote, it is not specified